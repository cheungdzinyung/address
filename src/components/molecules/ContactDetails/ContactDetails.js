import React from "react";
import { useLocation } from "react-router-dom";
import styles from "./ContactDetails.module.css";

export const ContactDetails = () => {
  const location = useLocation();
  const contact = location.state;

  return (
    <div className={styles.contactDetails}>
      {contact && (
        <>
          <img
            src={contact.picture.large}
            alt={`${contact.name.first} ${contact.name.last}`}
          />
          <h2>
            {contact.name.title}. {contact.name.first} {contact.name.last}
          </h2>
          <h3>Phone: {contact.phone} </h3>
          <h3>Cell: {contact.cell}</h3>
        </>
      )}
    </div>
  );
};
