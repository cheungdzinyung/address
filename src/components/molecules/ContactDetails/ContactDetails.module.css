.contactDetails {
  margin: 25px auto;
  padding: 75px;
  width: auto;
  display: inline-block;
  background-color: #d8d8d8;
  border: 1px solid #e7e7e7;
  box-shadow: 5px 5px 5px #ccc;
  color: #2d2d2d;
  border-radius: 20px;
}
