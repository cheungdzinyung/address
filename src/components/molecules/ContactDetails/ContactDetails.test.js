import React from "react";
import Router from "react-router-dom";
import { render, screen } from "@testing-library/react";
import { ContactDetails } from "./ContactDetails";

const mockUseLocationValue = {
  pathname: "/contact",
  search: "",
  hash: "",
  state: {
    name: "Andrew",
    picture: { large: "test" },
    phone: "phone123",
    cell: "cell123",
  },
};

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: jest.fn(),
}));

describe("ContactDetails", () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should render contact info", async () => {
    jest.spyOn(Router, "useLocation").mockReturnValue(mockUseLocationValue);
    const component = render(<ContactDetails />);

    const phoneElement = screen.getByText(/phone123/i);
    expect(phoneElement).toBeInTheDocument();

    const cellElement = screen.getByText(/cell123/i);
    expect(cellElement).toBeInTheDocument();

    expect(component).toMatchSnapshot();
  });
});
