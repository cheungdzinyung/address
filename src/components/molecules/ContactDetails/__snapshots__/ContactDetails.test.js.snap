// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetails should render contact info 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="contactDetails"
      >
        <img
          alt="undefined undefined"
          src="test"
        />
        <h2>
          . 
           
        </h2>
        <h3>
          Phone: 
          phone123
           
        </h3>
        <h3>
          Cell: 
          cel213
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="contactDetails"
    >
      <img
        alt="undefined undefined"
        src="test"
      />
      <h2>
        . 
         
      </h2>
      <h3>
        Phone: 
        phone123
         
      </h3>
      <h3>
        Cell: 
        cel213
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
